src/utilities.js

```diff
@@ -27,17 +29,21 @@ util.assertAtMostOne = function (collection, errorMsgPrefix) {
 
 /**
  *  Throws an exception if the data is not one of the expected types.
- * @param data the value to be checked
+ * @param data the value to be checked.  This may be a ResourceNode.
  * @param types an array of the permitted types
  * @param errorMsgPrefix An optional prefix for the error message to assist in
  *  debugging.
+ * @return the value that was checked.  If "data" was a ResourceNode, this will
+ *  be the ReourceNode's data.
  */
 util.assertType = function(data, types, errorMsgPrefix) {
```

Comment:
Given that the function now has dual-purposes, would you consider renaming it to something like assertTypeAndGet

Reply:
I thought about it, but I don't think that would make the meaning any clearer, unless added something after "Get" to say what was being gotten, and then it seems too long.